File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--add-a-filter-descriptor-to-a-finder-method_1.vb
    Public Shared Function ReadList(ByVal OrderDateParam As DateTime) As IEnumerable(Of SalesOrderHeader)
        Const ServerName As String = "MySQLServerName"
        Dim dataContext As AdventureWorksDataContext = _
            New AdventureWorksDataContext("Data Source=" & ServerName & _
                ";Initial Catalog=AdventureWorks;Integrated Security=True")

        Dim NoValuePassedIn As DateTime = Convert.ToDateTime("1/1/1900 12:00:00 AM")
        Dim DefaultDateTime As DateTime = Convert.ToDateTime("2001-09-01 00:00:00.000")

        ' If the user does not provide a value for the filter.
        If OrderDateParam = NoValuePassedIn Then
            ' Use a default date time value.
            OrderDateParam = DefaultDateTime
        End If

        Dim SalesOrderHeader As IEnumerable(Of SalesOrderHeader) = _
            From SalesOrderHeaders In dataContext.SalesOrderHeaders _
            Where SalesOrderHeaders.OrderDate = OrderDateParam _
            Select SalesOrderHeaders
        Return SalesOrderHeader
    End Function

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--add-a-filter-descriptor-to-a-finder-method_1.cs
        public static IEnumerable<SalesOrderHeader> ReadList(DateTime OrderDateParam)
        {
            const string ServerName = "MySQLServerName";
            AdventureWorksDataContext dataContext = new AdventureWorksDataContext
                  ("Data Source=" + ServerName + ";" +
                   "Initial Catalog=AdventureWorks;Integrated Security=True");

            DateTime NoValuePassedIn = Convert.ToDateTime("1/1/1900 12:00:00 AM");
            DateTime DefaultDateTime = Convert.ToDateTime("2001-09-01 00:00:00.000");

            // If the user does not provide a value for the filter.
            if (OrderDateParam == NoValuePassedIn)
            {
                // Use a default date time value.
                OrderDateParam = DefaultDateTime;
            }

            IEnumerable<SalesOrderHeader> SalesOrderHeader =
                from salesOrderHeaders in dataContext.SalesOrderHeaders
                where salesOrderHeaders.OrderDate == OrderDateParam
                select salesOrderHeaders;
            return SalesOrderHeader;
        }